app/Http/Controllers/EventController.php

```diff
@@ -15,222 +15,224 @@ class EventController extends Controller
+        }
+        if($request->hasFile('quizimage'))
+        {
+            $img = $request->file('quizimage');
+            $path_parts = pathinfo($_FILES["quizimage"]["name"]);
+            $image_path = $path_parts['filename'].'_'.time().'.'.$path_parts['extension'];
+            $request->quizimage->move(public_path('img'), $image_path);
+            $task->img = 'img/'.$image_path;
+        }
+        $task->start = $request->start_time;
+        $task->end = $request->end_time;
+        $task->duration = $request->duration;
+        $task->correctmark = $request->correct_mark;
+        $task->wrongmark = $request->wrong_mark;
+        $task->quedisplay = $request->display_ques;
+        $task->creator = auth::id();
+        $task->isactive = '0';
+
+        $task->save();
+
+        $eventIDs = session('TeacherEvent');
+        array_push($eventIDs,$task->id);
+        session(['TeacherEvent' => $eventIDs]);
+
+        return redirect('teacher/event/'.$task->id)->with('Event','Event added successfully');
+    }
+
+    public function add(Request $request, $id)
+    {
+        $this -> validate($request,
+            [
+                'question' => 'required|not_in:<br>',
+                'quetype' => 'required|digits_between:0,1',
+                'opt1' => 'required|not_in:<br>',
+                'opt2' => 'required|not_in:<br>',
+            ]
+            ,[
+                'not_in' => 'The :attribute field is required.',
+            ]
+        );
+        
+        $count = $request->count;
+        $flag=0;
+        for($y=1;$y <= $count; $y++){
+            $opti = $request->input('option'.$y);
+            if(!is_null($opti)){
+                $flag=1;
+                break;
+            }
+        }
+        if($flag==0)
+            return back()->with('Option', 'Please select atleast 1 correct answer')->withInput($request->all);
+        if($flag==0)
+            return back()->with('Option', 'Please select atleast 1 correct answer')->withInput($request->all);
+        
+        $addque = new Queans;
+        $addque->que = $request->question;
+        $addque->eventid = $id;
+        $addque->quetype = $request->quetype;
+        $addque->save();
+        
+        for($x=1; $x<=$count; $x++){
+            $option = new Option;
+            $option->queid = $addque->id;
+            $optans = $request->input('opt'.$x);
+            if(!is_null($optans) && $optans != "<br>"){
+                $option->ans = $optans;
+                $opt = $request->input('option'.$x);
+                if(is_null($opt))
+                    $option->iscorrect = 0;
+                else $option->iscorrect = 1;
+                $option->save();
+            }
+        }
+        return back()->with('success','Question added successfully.');
+    }
+
+    public function editEvent(Request $request, $id){
+        $event = Event::findOrFail($id);
```

Comment:
Every where you have used findOrFail, its good to use findOrFail but its adviced to use proper error handling by showing the message to the user that this event/whatever is invalid or not found rather showing 404 page for every error

Reply:
May I design a separate page for all bad request?

Reply:
You can check like this- `$sub = Subject::select('id')->where('id',$request->subject)->first();`
and return back with error message as you did before